src/app/api-server/src/dummy_auth_provider.rs

```diff
@@ -8,127 +8,81 @@
-use kamu::domain::auth;
+use dill::{component, interface};
+use kamu_accounts::*;
+use opendatafabric::{AccountID, AccountName};
 use serde::{Deserialize, Serialize};
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
+pub(crate) const DUMMY_AUTH_PROVIDER_IDENTITY_KEY: &str = "kamu-node";
+
+/////////////////////////////////////////////////////////s////////////////////////////////
+
 pub(crate) struct DummyAuthProvider {
-    predefined_accounts: HashMap<String, auth::AccountInfo>,
+    predefined_accounts: HashMap<String, AccountConfig>,
 }
 
+#[component(pub)]
+#[interface(dyn AuthenticationProvider)]
 impl DummyAuthProvider {
-    pub(crate) fn new_with_default_account() -> Self {
-        Self::new(vec![auth::AccountInfo {
-            account_id: opendatafabric::FAKE_ACCOUNT_ID.to_string(),
-            account_name: opendatafabric::AccountName::new_unchecked(auth::DEFAULT_ACCOUNT_NAME),
-            account_type: auth::AccountType::User,
-            display_name: String::from(auth::DEFAULT_ACCOUNT_NAME),
-            avatar_url: Some(String::from(auth::DEFAULT_AVATAR_URL)),
-            is_admin: true,
-        }])
-    }
-
-    pub(crate) fn new(predefined_accounts: Vec<auth::AccountInfo>) -> Self {
+    pub(crate) fn new(config: Arc<PredefinedAccountsConfig>) -> Self {
         Self {
-            predefined_accounts: predefined_accounts
-                .into_iter()
-                .map(|acc| (acc.account_name.to_string(), acc))
+            predefined_accounts: config
+                .predefined
+                .iter()
+                .map(|account| (account.account_name.to_string(), account.clone()))
                 .collect(),
         }
     }
 
-    fn find_account_info_impl(&self, account_name: &String) -> Option<auth::AccountInfo> {
-        // The account might be predefined in the configuration
-        self.predefined_accounts.get(account_name).cloned()
-    }
-
-    fn get_account_info_impl(
+    fn get_account(
         &self,
         account_name: &String,
-    ) -> Result<auth::AccountInfo, auth::RejectedCredentialsError> {
-        // The account might be predefined in the configuration
+    ) -> Result<AccountConfig, RejectedCredentialsError> {
         match self.predefined_accounts.get(account_name) {
-            // Use the predefined record
             Some(account_info) => Ok(account_info.clone()),
-
-            None => {
-                // Otherwise we don't recognized this user between predefined
-                Err(auth::RejectedCredentialsError::new(
-                    "Login of unknown accounts is disabled".to_string(),
-                ))
-            }
+            None => Err(RejectedCredentialsError {}),
         }
     }
 }
 
 #[async_trait::async_trait]
-impl auth::AuthenticationProvider for DummyAuthProvider {
-    fn login_method(&self) -> &'static str {
-        "password"
+impl AuthenticationProvider for DummyAuthProvider {
```

Comment:
Let's revise if we need any customization in `kamu-node` for the PASSWORD auth provider after recent refactorings. I have a feeling we might simply instantiate the implementation from `kamu-accounts-services` as is, and remove this customization, as predefined users configuration is now injected.